### PR TITLE
Replace FP16AltImpl with the detection of DataTypeA/B != DataType

### PR DESCRIPTION
**Pull request by Ray-Tw**

### CODE REVIEW

File: clients/gtest/matmul_gtest.yaml [hunk 1]
```diff
@@ -583,6 +583,22 @@ Tests:
...
+  N: [128, 129]
+  K: [128, 129]
+  transA_transB: *transA_transB_range
+  alpha: 1
+  beta: [ 0, 2 ]
+  bias_vector: 1
+  bias_type: [f32_r]
+  unit_check: 1
+  gpu_arch: '94?'
+
```
Comment (solaslin) [inline clients/gtest/matmul_gtest.yaml]: '94? ' is not correct. You are adding things on mi200, so you should make it '90a'

File: tensilelite/Tensile/Source/client/include/TypedId.hpp [hunk 2]
```diff
@@ -108,6 +108,7 @@ namespace Tensile
 #ifdef TENSILE_USE_BF16
     using TypedGemm_B_B_S = TypedGemm<BFloat16, BFloat16, BFloat16, BFloat16, float, float>;
     using TypedGemm_B_S_S = TypedGemm<BFloat16, BFloat16, float, float>;
+    using TypedGemm_HH_B_H_S = TypedGemm<Half, Half, Half, Half, float, float, BFloat16>;
 #endif // TENSILE_USE_BF16
 #ifdef TENSILE_USE_FP8_BF8
```
Comment (solaslin) [inline tensilelite/Tensile/Source/client/include/TypedId.hpp]: I'm not sure of this naming, maybe @jichangjichang you have any comments?

File: clients/gtest/matmul_gtest.yaml (hunk 1 above)
Comment (Ray-Tw) [reply clients/gtest/matmul_gtest.yaml]: It has been updated, thanks!

File: tensilelite/Tensile/Source/client/include/TypedId.hpp (hunk 2 above)
Comment (jichangjichang) [reply tensilelite/Tensile/Source/client/include/TypedId.hpp]: better to be H_B_H_S

File: library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aldebaran/104CU/GridBased/aldebaran_Cijk_Ailk_Bjlk_HH_BHS_BH_Bias_AS_SAV.yaml [hunk 3]
```diff
@@ -0,0 +1,613 @@
...
+- [Device 0049, Device 0050]
+- Activation: true
+  ActivationComputeDataType: 0
+  ActivationNoGuard: false
+  ActivationType: all
+  AllowNoFreeDims: false
+  AssignedDerivedParameters: true
+  Batched: true
+  BetaOnlyUseBias: false
+  BiasDataTypeList: [0, 4, 7]
```
Comment (jichangjichang) [inline library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aldebaran/104CU/GridBased/aldebaran_Cijk_Ailk_Bjlk_HH_BHS_BH_Bias_AS_SAV.yaml]: Default Bias Type should be 0(accumulation type ),4 (output type)

**Comment by jichangjichang** [comment]
The logical yaml you uploaded is only for NT. Please support other 3 transpose as well

**Comment by KKyang** [comment]
Please enable SupportUserArgs: True if possible.

**Comment by Ray-Tw** [comment]
> Please enable SupportUserArgs: True if possible.

Should it provide both SupportUserArgs=**True** and SupportUserArgs=**False**, or just SupportUserArgs=**True** ?

**Comment by KKyang** [comment]
> > Please enable SupportUserArgs: True if possible.
> 
> Should it provide both SupportUserArgs=**True** and SupportUserArgs=**False**, or just SupportUserArgs=**True** ?

Just True

**Comment by Ray-Tw** [comment]
@jichangjichang 
> better to be H_B_H_S
> The logical yaml you uploaded is only for NT. Please support other 3 transpose as well

@KKyang 
> Please enable SupportUserArgs: True if possible.

Updated, thanks!

File: library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aldebaran/104CU/GridBased/aldebaran_Cijk_Alik_Bjlk_HH_BHS_BH_BiasSH_AS_SAV_UserArgs.yaml [hunk 4]
```diff
@@ -0,0 +1,613 @@
...
+    _UseSgprForGRO: false
+    _VectorStore: 1
+    _WorkspaceSizePerElemBias: 0
+    _WorkspaceSizePerElemC: 4
+    _staggerStrideShift: 1
+  - 1LDSBuffer: 0
+    ActivationAlt: false
+    ActivationFuncCall: true
+    ActivationFused: true
+    AssertFree0ElementMultiple: 8
```
Comment (jichangjichang) [inline library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aldebaran/104CU/GridBased/aldebaran_Cijk_Alik_Bjlk_HH_BHS_BH_BiasSH_AS_SAV_UserArgs.yaml]: Gridbase is common solution pool. We should use AssertXXXElementMultiple: 1 for these common solutions

**Comment by jichangjichang** [comment]
Please remember to remove "Fp16AltImpl: false" from all other yamls.

**Comment by Ray-Tw** [comment]
> Please remember to remove "Fp16AltImpl: false" from all other yamls.

Updated, thanks!

Done : Add yaml for 110CU

**Review by KKyang — APPROVED** [review]

**Comment by jichangjichang** [comment]
LibraryIO.py and matmul_gtest.yaml still contains "Fp16AltImpl" and "fp16altimpl". Please remove it if possible.

**Comment by Ray-Tw** [comment]
> LibraryIO.py and matmul_gtest.yaml still contains "Fp16AltImpl" and "fp16altimpl". Please remove it if possible.

LibraryIO.py/matmul_gtest.yaml and all logic yamls are updated, thanks!

**Review by jichangjichang — APPROVED** [review]
lgtm

File: clients/include/testing_matmul.hpp [hunk 5]
```diff
@@ -1814,7 +1814,7 @@ void testing_matmul(const Arguments& arg)
             auto scaleBValue = arg.scaleB ? (*hScaleB[gemmIdx])[0] : 1;
             auto scaleDValue = arg.scaleD ? (*hScaleD[gemmIdx])[0] : 1;
             auto scaleEValue = arg.scaleE ? (*hScaleE[gemmIdx])[0] : 1;
-
+
```
Comment (jichangjichang) [inline clients/include/testing_matmul.hpp]: remove this

Comment (Ray-Tw) [reply clients/include/testing_matmul.hpp]: Updated, thanks!

**Pull request merged**